pkg/cluster/deploybaseresources_additional.go

```diff
@@ -92,6 +92,20 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
+func (m *manager) fpspStorageBlobContributorRBAC(storageAccountName string) (*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, fmt.Errorf("fpspStorageBlobContributorRBAC called for a CSP cluster")
+	}
+	resourceTypeStorageAccount := "Microsoft.Storage/storageAccounts"
+	return rbac.ResourceRoleAssignmentWithName(
+		rbac.RoleStorageBlobDataContributor,
+		"'"+m.env.FPServicePrincipalID()+"'",
```

Comment:
I don't think this is going to work.

We should be using MSGraph in the customer's tenant to look up the customer's service principal backing the 1p application.  In this case, we're going to be granting our azure managed environment service principal object_id/principal_id permissions, which doesn't exist in the customer's tenant. 

 
`m.env.FPServicePrincipalID()` should be our local tenant service principal.  You want to use the remote/customer's tenant service principal, which we'll have to look up using MSGraph, or it might be included in the token we get in the 1p application when we authenticate.

Reply:
Our checkaccessv2 validation pulls the OID of the 1p app in the customer's tenant.  That's how you can do this here.


```
		oid, err := token.GetObjectId(t.Token)
		if err != nil {
			c.dv.log.Error("Unable to parse the token oid claim: ", err)
			return false, err
		}

Reply:
Thanks for pointing out this, changed the flow to get the fpspID when the fpspClusterTenantCred is created.